common/service-support-cdi/src/main/java/io/helidon/common/servicesupport/cdi/CdiExtensionBase.java

```diff
@@ -0,0 +1,483 @@
+ *     {@code Suspended} {@code AsyncResponse}
+ *     parameter. It records information about those as instances of concrete subclasses of {@link AsyncResponseInfo}. These
+ *     info instances are collected inside a REST endpoint info data structure which extends {@link RestEndpointInfo}. Both of
+ *     these types are parameters to this class so concrete implementations can add data that is specific to their specific
+ *     technologies to the data structures. Concrete implementations of this class provide factory methods for these
+ *     parameterized types.
+ * </p>
+ *
+ * @param <A> concrete {@code AsyncResponseInfo} type
+ * @param <R> concrete {@code RestEndpointInfo} type
+ * @param <T> concrete type of {@code ServiceSupportBase} used
+ * @param <B> Builder for the concrete type of {@code }ServiceSupportBase}
+ */
+public abstract class CdiExtensionBase<
+        A extends CdiExtensionBase.AsyncResponseInfo,
+        R extends CdiExtensionBase.RestEndpointInfo,
+        T extends ServiceSupportBase<T, B>,
+        B extends ServiceSupportBase.Builder<T, B>> implements Extension {
+    private final Map<Bean<?>, AnnotatedMember<?>> producers = new HashMap<>();
+
+    private final Set<Class<?>> annotatedClasses = new HashSet<>();
+    private final Set<Class<?>> annotatedClassesProcessed = new HashSet<>();
+    private final Set<Class<? extends Annotation>> annotations;
+
+    private final Logger logger;
+    private final Class<?> ownProducer;
+    private final Function<Config, T> serviceSupportFactory;
+    private final String configPrefix;
+
+    private T serviceSupport = null;
+
+    private R restEndpointInfo = null;
+
+    /**
+     * Common initialization for concrete implementations.
+     *
+     * @param logger Logger instance to use for logging messages
+     * @param annotations set of annotations this extension handles
+     * @param ownProducer type of producer class use in creating beans needed by the extension
+     * @param serviceSupportFactory function from config to the corresponding SE-style service support object
+     * @param configPrefix prefix for retrieving config related to this extension
+     */
+    protected CdiExtensionBase(
+            Logger logger,
+            Set<Class<? extends Annotation>> annotations,
+            Class<?> ownProducer,
+            Function<Config, T> serviceSupportFactory,
+            String configPrefix) {
+        this.logger = logger;
+        this.annotations = annotations;
+        this.ownProducer = ownProducer; // class containing producers provided by this module
+        this.serviceSupportFactory = serviceSupportFactory;
+        this.configPrefix = configPrefix;
+    }
+
+    /**
+     * Returns the real class of this object, skipping proxies.
+     *
+     * @param object The object.
+     * @return Its class.
+     */
+    public static Class<?> getRealClass(Object object) {
```

Comment:
Do not use `get` - even when refactoring from existing classes, as this method was most likely private. If it is public, it becomes part of Helidon public API and must follow rules of Helidon API design.

Reply:
Fixed